pkg/remotecmd/kubeexec.go

```diff
@@ -318,40 +318,76 @@ func (k *kubeExecProcessHandler) getProcessInfoFromPid(
 	return process, nil
 }
 
+// getAllProcesses returns a map of all the processes and their direct children:
+// i) the key is the process PID;
+// ii) and the value is a list of all its direct children.
+// It does so by reading all /proc/<pid>/stat files. More details on https://man7.org/linux/man-pages/man5/proc.5.html.
+func (k *kubeExecProcessHandler) getAllProcesses(podName string, containerName string) (map[int][]int, error) {
+	stdout, stderr, err := k.execClient.ExecuteCommand([]string{ShellExecutable, "-c", "cat /proc/*/stat || true"},
```

Comment:
> Why did you remove err propagation 2>/dev/null? Is it because it will be returned separately via `stderr`?
>
Yes, I removed `2 > /dev/null` to not lose any useful information that might end up in `stderr`;  I thought it could be helpful to display both `stderr` and `stdout` if `err != nil`.

>  Also, I am still confused by what `|| true` does.
>
`cmd1 || cmd2` means, if `cmd1` fails with non-zero exit code, execute `cmd2` and return the exit code of `cmd2`, otherwise return 0 (exit code of `cmd1`). So ` cmd || true` is a way to always return 0 as the exit code (so making `cmd` look as if it succeeded), regardless of the actual exit code of `cmd`.
In our case, if we don't return 0, the `exec` call we are doing will interpret this as an error; this could be the case since not all `/proc/*` folders have a `stat` file. For example:

```shell
$ kubectl exec -it pid-nodejs-app-55cbc55b5-sb42p -- /bin/sh -c 'cat /proc/*/stat > /dev/null'
cat: /proc/net/stat: Is a directory
command terminated with exit code 1

$ echo $?
1
```

Reply:
Thank you for explaining that :)